### ISSUE TITLE
Racist neural network?

### ISSUE THREAD

**Issue by iperov** [adamian98/pulse#2]
LOL
![photo_2020-06-13_18-21-52](https://user-images.githubusercontent.com/8076202/84571228-f7af7600-ada2-11ea-9036-ec9bcf076f47.jpg)
I like it.

**Comment by sachit-menon** [comment]
Hi,

This is an important issue and one we take seriously. The reason this kind of behavior sometimes comes up stems from the inherent biases in the pretrained generative model we used, StyleGAN, and the biases it inherits from the dataset it was trained on, Flickr Faces HQ (FFHQ). For an in depth study about the biases seen in StyleGAN, see Salminen et al, 2020. Per the FFHQ documentation: “The images were crawled from Flickr <https://www.flickr.com/>, thus inheriting all the biases of that website.” 

We personally didn’t have the resources to create a new dataset or train a large-scale generative model at the level of NVIDIA. However, the general approach to PULSE works with any generative model. We are always on the lookout for less biased generative models. If you’re aware of a better alternative to StyleGAN, please let us know.

Salminen, Joni, et al. “Analyzing Demographic Bias in Artificially Generated Facial Pictures.” Extended Abstracts of the 2020 CHI Conference on Human Factors in Computing Systems, ACM, 2020, pp. 1–8. DOI.org (Crossref), doi:10.1145/3334480.3382791.

NVlabs/Ffhq-Dataset. 2019. NVIDIA Research Projects, 2020. GitHub, https://github.com/NVlabs/ffhq-dataset.

**Comment by iperov** [comment]
> the general approach to PULSE works with any generative model

but such model must not know about blurry faces?
because why to produce sharp HR faces, if downscaled blurry HR faces will produce equal LR ?

**Comment by Drumshtick** [comment]
![SLJ](https://user-images.githubusercontent.com/38117121/84576052-02501800-ad6f-11ea-8249-176c1baf7eeb.png)

**Comment by adamian98** [comment]
> > the general approach to PULSE works with any generative model
> 
> but such model must don't know about blurry faces?
> because why to produce sharp HR faces, if downscaled blurry HR faces will produce equal LR ?

Yes, we always assume that the generator parameterizes the space of high resolution natural looking images. If the generator is capable of producing blurry images then PULSE is as well.

**Comment by stefan-reich** [comment]
No it's not racist, it just has a certain view of reality.

**Comment by iperov** [comment]
@sachit-menon the topic is joke. Nevermind. Black lives don't matter :D

**Comment by stefan-reich** [comment]
@the guys with the thumb downs: You actually think a network can be "racist"? That's just ridiculous.

**Comment by doantientai** [comment]
> @sachit-menon the topic is joke. Nevermind. Black lives don't matter :D

Every single live matters and this is a wonderful project.

The results of the model reflect the training data distribution. It works as expected. If you can collect a more robust data then I believe the result will be more diverse.

**Comment by stefan-reich** [comment]
> The results of the model reflect the training data distribution. It works as expected. If you can collect a more robust data then I believe the result will be more diverse.

I pretty much commented the same thing, but it was thumbed down, so I deleted it. I basically said that the network just has a certain view of reality (which it does). Guess I failed to acknowledge the PC label "racist" that has to be tacked onto everything today.
